platform-dom/src/main/java/com/softicar/platform/dom/elements/input/auto/DomAutoCompleteDefaultInputEngine.java

```diff
@@ -0,0 +1,208 @@
+package com.softicar.platform.dom.elements.input.auto;
+
+import com.softicar.platform.common.container.derived.DerivedObject;
+import com.softicar.platform.common.core.i18n.IDisplayString;
+import com.softicar.platform.common.core.i18n.IDisplayable;
+import com.softicar.platform.common.core.locale.CurrentLocale;
+import com.softicar.platform.common.core.utils.CastUtils;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+/**
+ * Default implementation of {@link IDomAutoCompleteInputEngine}.
+ * <p>
+ * To determine all valid elements, this implementation relies on a
+ * {@link Supplier} defined by calling {@link #setLoader}. Internally, these
+ * loaded elements are then cached together with their respective
+ * {@link IDisplayString}, using {@link DerivedObject}. Call
+ * {@link #addDependsOn} to define dependencies to control cache invalidation.
+ * <p>
+ * By default, the {@link IDisplayString} for each element is determined by
+ * testing if the element implements the interface {@link IDisplayable}. If this
+ * fails, the {@link Object#toString} method is used as a fallback. This
+ * mechanism can be overridden by calling {@link #setDisplayFunction(Function)}.
+ * <p>
+ * Elements evaluating to equal {@link IDisplayString} instances are
+ * deduplicated by appending appropriate suffixes. The comparison is done
+ * case-insensitive, although the letter cases of the respective
+ * {@link IDisplayString} instances are retained, e.g. "foo" and "Foo" will
+ * become "foo (1)" and "Foo (2)".
```

Comment:
I think, in addition to being case-insensitive, the comparison also needs to be diacritic-insensitive.
Let there be two entities, a and b, with the following display names:
- a: "Bar"
- b: "Bǎr" 

That's a https://en.wikipedia.org/wiki/Caron diacritic. Now, try to select b with an English or German keyboard layout. Yeah...
Btw, we already have `DiacriticNormalizer` for that.

Reply:
- yes
- yes
- yes

Reply:
ok PLAT-1032 created (and related PLAT-1033)